layouts/tech-partner-form/single.html

```diff
@@ -0,0 +1,177 @@
+      }
+    }
+
+
+    backButton.onclick = function (event) {
+      // no back button on first step
+      if(index == 1 && backButton) {
+        backButton.setAttribute("style", "display:none");
+      }
+      parent[index].setAttribute("style", "display:none;");
+      if(progressBar) {
+        progressBar.innerHTML = index
+        progressBarValue = progressBarValue - 8.333;
+        progressBar.setAttribute("style", `width: ${progressBarValue}%`);
+      }
+      index = index - 1
+      parent[index].setAttribute("style", "display:block;");
+    }
+
+    submitButton.onclick = function (event) {
+      var userInput = document.querySelectorAll(".user-input") // all non-selecter user inputs
+
+      for (i = 0; i < userInput.length; i++) {
+        partner_form[userInput[i].name] = userInput[i].value
+      }
+
+      var useCaseOptions = document.getElementsByClassName('use-case');
+
+      if(useCaseOptions) {
+        partner_form.use_case = [];
+        for (var i = 0, length = useCaseOptions.length; i < length; i++) {
+          if (useCaseOptions[i].selected) {
+            partner_form.use_case.push(useCaseOptions[i].value);
+          }
+        }
+      }
+
+      var dataAmountOptions = document.getElementsByClassName('data-amount');
+
+      if(dataAmountOptions) {
+        for (var i = 0, length = dataAmountOptions.length; i < length; i++) {
+          if (dataAmountOptions[i].selected) {
+             partner_form.data_amount = dataAmountOptions[i].value;
+             break;
+          }
+        }
+      }
+
+      var fileSizeOptions = document.getElementsByClassName('file-size');
+
+      if(fileSizeOptions) {
+        for (var i = 0, length = fileSizeOptions.length; i < length; i++) {
+          if (fileSizeOptions[i].selected) {
+           partner_form.file_sizes = fileSizeOptions[i].value;
+           break;
+          }
+        }
+      }
+
+      var perMonthOptions = document.getElementsByClassName('per-month');
+
+      if(perMonthOptions) {
+        for (var i = 0, length = perMonthOptions.length; i < length; i++) {
+          if (perMonthOptions[i].selected) {
+             partner_form.download_times_per_month = perMonthOptions[i].value;
+             break;
+           }
+        }
+      }
+
+      // timestamps with timezone
+      partner_form.timestamp = Math.floor((new Date()).getTime() / 1000)
+
+
+      analytics.identify(analytics.user().anonymousId(),
+        {email: partner_form.email, timestamp: partner_form.timestamp, partner_form: true},
+        {integrations: {'All': false, 'Customer.io': true}}, function() {
+          analytics.track("tech-partner-form", {});
+      });
+
+      analytics.identify(analytics.user().anonymousId(),
+        partner_form,
+        {integrations: {'All': false, 'postgresSelfHosted': true}}, function() {
+          window.location = "/tech-partner-thank-you"
+      });
+    }
+
+  }
```

Comment:
please refactor this huge function using many small functions, where their names will tell us what the exact part of code is doing. Because it's almost impossible to imagine what is going on here.

Like here https://github.com/storj/storj/blob/master/web/satellite/static/resetPassword/resetPassword.js